src/transformers/generation/utils.py

```diff
@@ -454,20 +454,31 @@ def _maybe_initialize_input_ids_for_generation(
     def _prepare_attention_mask_for_generation(
         self,
         inputs: torch.Tensor,
-        pad_token_id: Optional[int],
-        eos_token_id: Optional[Union[int, List[int]]],
+        pad_token_id: Optional[torch.Tensor],
+        eos_token_id: Optional[torch.Tensor],
     ) -> torch.LongTensor:
+        # No information for attention mask inference -> return default attention mask
```

Comment:
The logic rewritten in functions like this is `torch.compile(..., fullgraph=True)` compatible 😉